<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [Mono-bugs] [Bug 665241] New: System.Data.Common.DataAdapter throws IndexOutOfRangeException
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:mono-bugs%40lists.ximian.com?Subject=%5BMono-bugs%5D%20%5BBug%20665241%5D%20New%3A%20System.Data.Common.DataAdapter%20throws%0A%20IndexOutOfRangeException&In-Reply-To=">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="108428.html">
   <LINK REL="Next"  HREF="108601.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[Mono-bugs] [Bug 665241] New: System.Data.Common.DataAdapter throws IndexOutOfRangeException</H1>
    <B>bugzilla_noreply at novell.com</B> 
    <A HREF="mailto:mono-bugs%40lists.ximian.com?Subject=%5BMono-bugs%5D%20%5BBug%20665241%5D%20New%3A%20System.Data.Common.DataAdapter%20throws%0A%20IndexOutOfRangeException&In-Reply-To="
       TITLE="[Mono-bugs] [Bug 665241] New: System.Data.Common.DataAdapter throws IndexOutOfRangeException">bugzilla_noreply at novell.com
       </A><BR>
    <I>Tue Jan 18 14:53:05 EST 2011</I>
    <P><UL>
        <LI>Previous message: <A HREF="108428.html">[Mono-bugs] [Bug 665233] [PATCH] ToolStrip Menus Keeping Keyboard Capture after hovering over SubMenu.
</A></li>
        <LI>Next message: <A HREF="108601.html">[Mono-bugs] [Bug 665241] System.Data.Common.DataAdapter throws IndexOutOfRangeException
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#108143">[ date ]</a>
              <a href="thread.html#108143">[ thread ]</a>
              <a href="subject.html#108143">[ subject ]</a>
              <a href="author.html#108143">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>
<A HREF="https://bugzilla.novell.com/show_bug.cgi?id=665241">https://bugzilla.novell.com/show_bug.cgi?id=665241</A>

<A HREF="https://bugzilla.novell.com/show_bug.cgi?id=665241#c0">https://bugzilla.novell.com/show_bug.cgi?id=665241#c0</A>


           Summary: System.Data.Common.DataAdapter throws
                    IndexOutOfRangeException
    Classification: Mono
           Product: Mono: Class Libraries
           Version: 2.8.x
          Platform: VMWare
        OS/Version: openSUSE 11.3
            Status: NEW
          Severity: Normal
          Priority: P5 - None
         Component: Sys.Data
        AssignedTo: <A HREF="http://lists.ximian.com/mailman/listinfo/mono-bugs">mono-bugs at lists.ximian.com</A>
        ReportedBy: <A HREF="http://lists.ximian.com/mailman/listinfo/mono-bugs">andorbal at gmail.com</A>
         QAContact: <A HREF="http://lists.ximian.com/mailman/listinfo/mono-bugs">mono-bugs at lists.ximian.com</A>
          Found By: ---
           Blocker: No


Description of Problem:
While testing my application on mono, I ran into this exception.  It seems that
if a column is specified twice in a SELECT statement using different cased
letters, the IndexOutOfRangeException is thrown.  The SELECT works fine using a
DataReader but fails using a DataAdapter to fill a DataTable.  I am using
Microsoft SQL Server, so I can only verify that the bug occurs using it.

Steps to reproduce the problem:
1. Create a table in SQL Server using the following SQL:
CREATE TABLE [dbo].[AccountsTest](
    [CompanyName] [varchar](100) NULL
)

2. Build a simple application with the following code:
Console.Write(&quot;Trying with identical case: &quot;)

        Dim sqlCommSucceeds As New SqlCommand(&quot;SELECT Companyname, Companyname
FROM AccountsTest&quot;, New SqlConnection(&lt;&lt;your_connection_string&gt;&gt;))
        Dim sqlAdapterSucceeds As New SqlDataAdapter(sqlCommSucceeds)

        Try
            sqlCommSucceeds.Connection.Open()
            Dim table As New DataTable
            sqlAdapterSucceeds.Fill(table)
            Console.WriteLine(&quot;Success!&quot;)
        Finally
            sqlCommSucceeds.Connection.Dispose()
            sqlCommSucceeds.Dispose()
        End Try


        Console.WriteLine(&quot;Trying with different case: &quot;)

        Dim sqlCommFails As New SqlCommand(&quot;SELECT CompanyName, Companyname
FROM AccountsTest&quot;, New SqlConnection(&lt;&lt;your_connection_string&gt;&gt;))
        Dim sqlAdapterFails As New SqlDataAdapter(sqlCommFails)

        Try
            sqlCommFails.Connection.Open()
            Dim table As New DataTable
            sqlAdapterFails.Fill(table)
            Console.WriteLine(&quot;Success!&quot;)    ' Never gets here -- an
IndexOutOfRangeException is thrown!
        Finally
            sqlCommFails.Connection.Dispose()
            sqlCommFails.Dispose()
        End Try


Actual Results:
Trying with identical case: Success!
Trying with different case:

Unhandled Exception: System.IndexOutOfRangeException: Array index is out of
range.
  at System.Data.Common.DataAdapter.FillTable (System.Data.DataTable dataTable,
IDataReader dataReader, Int32 startRecord, Int32 maxRecords, System.Int32&amp;
counter) [0x00000] in &lt;filename unknown&gt;:0 
  at System.Data.Common.DataAdapter.FillInternal (System.Data.DataTable
dataTable, IDataReader dataReader) [0x00000] in &lt;filename unknown&gt;:0 
  at System.Data.Common.DataAdapter.Fill (System.Data.DataTable dataTable,
IDataReader dataReader) [0x00000] in &lt;filename unknown&gt;:0 
  at System.Data.Common.DbDataAdapter.Fill (System.Data.DataTable dataTable,
IDbCommand command, CommandBehavior behavior) [0x00000] in &lt;filename unknown&gt;:0 
  at System.Data.Common.DbDataAdapter.Fill (System.Data.DataTable dataTable)
[0x00000] in &lt;filename unknown&gt;:0 
  at (wrapper remoting-invoke-with-check) System.Data.Common.DbDataAdapter:Fill
(System.Data.DataTable)
  at TestApplication.Main.Main () [0x00000] in &lt;filename unknown&gt;:0 

Expected Results:
Trying with identical case: Success!
Trying with identical case: Success!

How often does this happen? 
This happens every time the SELECT statement is run with the above conditions.

Additional Information:
I purposely used different casing in the successful SELECT statement than what
the column is defined as to show that the problem doesn't seem to be related to
that.  I realize that fixing the case and/or only including the column once
would fix the problem, but I think this is unexpected behavior and would be
tough for a developer to find why the exception is getting thrown.

-- 
Configure bugmail: <A HREF="https://bugzilla.novell.com/userprefs.cgi?tab=email">https://bugzilla.novell.com/userprefs.cgi?tab=email</A>
------- You are receiving this mail because: -------
You are the QA contact for the bug.
You are the assignee for the bug.
</PRE>











































<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="108428.html">[Mono-bugs] [Bug 665233] [PATCH] ToolStrip Menus Keeping Keyboard Capture after hovering over SubMenu.
</A></li>
	<LI>Next message: <A HREF="108601.html">[Mono-bugs] [Bug 665241] System.Data.Common.DataAdapter throws IndexOutOfRangeException
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#108143">[ date ]</a>
              <a href="thread.html#108143">[ thread ]</a>
              <a href="subject.html#108143">[ subject ]</a>
              <a href="author.html#108143">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="http://lists.ximian.com/mailman/listinfo/mono-bugs">More information about the mono-bugs
mailing list</a><br>
</body></html>
